rocAL/rocAL/rocAL_hip/rali_hip_kernels.cpp

```diff
@@ -0,0 +1,295 @@
+        float *out_tensor = (float *)output_tensor + n*img_offset + dst_buf_offset;
+        if (nchw.y == 3){
+            float3 dst;
+            if (reverse_channels) 
+                dst = make_float3((float)inp_img[srcIdx+2], (float) inp_img[srcIdx+1], (float)inp_img[srcIdx])*multiplier + offset;
+            else
+                dst = make_float3((float)inp_img[srcIdx], (float) inp_img[srcIdx+1], (float)inp_img[srcIdx+2])*multiplier + offset;
+            out_tensor[dstIdx] = dst.x;
+            out_tensor[dstIdx+W*H] = dst.y;
+            out_tensor[dstIdx+W*H*2] = dst.z;
+        } else{
+            out_tensor[dstIdx] = (float)inp_img[srcIdx]*multiplier.x + offset.x;
+        }
+    }
+}
+
+__global__ void __attribute__((visibility("default")))
+Hip_CopyInt8ToNCHW_fp16
+(
+    const uchar*     inp_image_u8,
+    void*     output_tensor,
+    unsigned int     dst_buf_offset,
+    uint4 nchw,
+    float3 multiplier,
+    float3 offset,
+    const unsigned int reverse_channels
+)
+{
+    int x = hipBlockDim_x * hipBlockIdx_x + hipThreadIdx_x;
+    int y = hipBlockDim_y * hipBlockIdx_y + hipThreadIdx_y;
+    const int W = nchw.w; const int H = nchw.z; 
+    const int img_offset = nchw.y * W * H;
+
+    if ((x >= W) || (y >= H)) return;
+    for (unsigned int n=0; n < nchw.x; n++) {
+        unsigned short *out_tensor = (unsigned short *)output_tensor + n*img_offset+dst_buf_offset;
+        const uchar *inp_img = &inp_image_u8[n*img_offset+dst_buf_offset];
+        unsigned int srcIdx =  y*W + x;
+        // copy float3  pixels to dst
+        unsigned int dstIdx =  y*W + x;
+        if (nchw.y == 3){
+            float3 dst;
+            if (reverse_channels) 
+                dst = make_float3((float)inp_img[srcIdx+2], (float) inp_img[srcIdx+1], (float)inp_img[srcIdx])*multiplier + offset;
+            else
+                dst = make_float3((float)inp_img[srcIdx], (float) inp_img[srcIdx+1], (float)inp_img[srcIdx+2])*multiplier + offset;
+            out_tensor[dstIdx] = __float2half(dst.x);
+            out_tensor[dstIdx+W*H] = __float2half(dst.y);
+            out_tensor[dstIdx+W*H*2] = __float2half(dst.z);
+        } else{
+            out_tensor[dstIdx] = __float2half((float)inp_img[srcIdx]*multiplier.x + offset.x);
+        }
+    }
+}
+
+int HipExecCopyInt8ToNHWC
+(
+    hipStream_t stream,
+    const void*     inp_image_u8,
+    void*     output_tensor,
+    unsigned int     dst_buf_offset,
+    const unsigned int     n,
+    const unsigned int     c,
+    const unsigned int     h,
+    const unsigned int     w,
+    float     multiplier0,
+    float     multiplier1,
+    float     multiplier2,
+    float     offset0,
+    float     offset1,
+    float     offset2,
+    unsigned int reverse_channels,
+    unsigned int fp16
+)
+{
+    int localThreads_x = 16, localThreads_y = 16;
+    int globalThreads_x = w, globalThreads_y = h;
+#if ENABLE_EVENT_BASED_SYNC   
```

Comment:
remove the code for ENABLE_EVENT_BASED_SYNC since it won't be used

Reply:
Done.